File: golden-paths/go/fiber/deliberate_test.go
//go:build deliberate

// Deliberately-failing test behind the `deliberate` build tag — proves the go lane PROPAGATES failure.
// A normal `go test ./...` cannot see it; the lane self-check runs `go test -tags deliberate -run DeliberateFailure`.
package main

import "testing"

func TestDeliberateFailure(t *testing.T) {
	t.Fatal("selfcheck: the golden-path go lane must surface this failure (exit non-zero)")
}

File: golden-paths/cpp/httplib/test/deliberate_test.cpp
#include <doctest/doctest.h>

// Deliberately-failing test — proves the cpp lane PROPAGATES failure. It lives in the "selfcheck" test
// suite; a normal run uses --test-suite-exclude=selfcheck and the lane self-check uses
// --test-suite=selfcheck (only it). Suite-based (not a name filter) → fail-closed on a rename: a removed
// tag makes --test-suite=selfcheck match zero cases → doctest exits 0 → the guard reports LANE BROKEN.
TEST_SUITE("selfcheck") {
  TEST_CASE("deliberate failure") {
    FAIL("selfcheck: the golden-path cpp lane must surface this failure (exit non-zero)");
  }
}
